src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2297,7 +2296,7 @@ def _if_construct_handler(self, node, parent):
                                    nodes=[clause.items[0]])
 
                 # Create if-body as second child
-                ifbody = Schedule(parent=ifblock)
+                ifbody = Schedule(parent=newifblock)
```

Comment:
I have added appropriate asserts now.

But also the new is_orphan check wouldn't let it pass anymore because if is not an orphan it must at least be pointing to the same parent that from where it is added.